js/scenes/Compass.js

```diff
@@ -21,27 +30,17 @@ export default class Debug extends Component {
     };
   }
 
-  componentDidMount() {
-    navigator.geolocation.getCurrentPosition((position) => {
-      var initialPosition = JSON.stringify(position);
-      this.setState({ initialPosition });
+  componentWillReceiveProps({ location }) {
+    if (!location) {
+      return;
     }
-      , (error) => console.log(JSON.stringify(error)),
-      { enableHighAccuracy: true, timeout: 20000, maximumAge: 1000 }
-    );
-    this.watchID = navigator.geolocation.watchPosition((position) => {
-      this.setState({
-        latitude: position.coords.latitude,
-        longitude: position.coords.longitude,
-      });
+    
+    this.setState({
+      latitude: location.value.coords.latitude,
+      longitude: location.value.coords.longitude,
```

Comment:
I can't see where `this.state.latitude` or `this.state.longitude` are being used.